scri/bms_transformations.py

```diff
@@ -0,0 +1,829 @@
+
+    def __repr__(self):
+        Lorentz_output = {}
+        for transformation in self.order:
+            if transformation == 'rotation':
+                Lorentz_output[transformation] = self.rotation
+            elif transformation == 'boost':
+                Lorentz_output[transformation] = self.boost
+
+        return f'Lorentz_transformation(\n\t{self.order[0]}={Lorentz_output[self.order[0]]}\n\t{self.order[1]}={Lorentz_output[self.order[1]]}\n)'
+
+    def __str__(self):
+        Lorentz_output = {}
+        for transformation in self.order:
+            if transformation == 'rotation':
+                Lorentz_output[transformation] = self.rotation
+            elif transformation == 'boost':
+                Lorentz_output[transformation] = self.boost
+
+        return f'Lorentz_transformation(\n\t{self.order[0]}={Lorentz_output[self.order[0]]}\n\t{self.order[1]}={Lorentz_output[self.order[1]]}\n)'
+    
+    def copy(self):
+        return Lorentz_transformation(rotation=self.rotation,
+                                      boost=self.boost,
+                                      order=self.order,
+                                      ell_max=self.ell_max)
+
+    def reorder(self, output_order):
+        """Reorder a Lorentz transformation.
+        """
+        if not ('rotation' in output_order and\
+                'boost' in output_order):
+            raise ValueError('Not enough transformations')
+        
+        if self.order == output_order:
+            return self
+        else:
+            L_spin_matrix = Lorentz_to_spin_matrix(self)
+            L_reordered = spin_matrix_to_Lorentz(L_spin_matrix, output_order=output_order)
+            return L_reordered
+
+    def is_identity(self, verbose=False):
+        """Check if a BMS transformation is the identity element
+        """
+        rotation_is_identity = np.allclose(self.rotation.components, [1,0,0,0])
+        boost_is_identity = np.allclose(self.boost, [0,0,0])
+        if verbose:
+            print(f"rotation is identity: {rotation_is_identity}")
+            print(f"boost is identity: {boost_is_identity}")
+        return rotation_is_identity and boost_is_identity
+
+    def inverse(self, output_order=None):
+        """Compute the inverse of a Lorentz transformation.
+        """
+        if output_order is None:
+            output_order = self.order[::-1]
+        
+        L_spin_matrix = Lorentz_to_spin_matrix(self)
+        L_spin_matrix_inverse = np.linalg.inv(L_spin_matrix)
+        L_inverse = spin_matrix_to_Lorentz(L_spin_matrix_inverse, output_order=output_order)
+        return L_inverse
+
+    def __eq__(self, other):
+        rotation_eq = np.allclose(self.rotation.components, other.rotation.components)
+        boost_eq = np.allclose(self.boost, other.boost)
+        return rotation_eq and boost_eq
+            
+    def compose(self, other, output_order=['rotation','boost']):
+        """Compose two Lorentz transformations.
```

Comment:
I think it would be useful here to say whether this is left- or right-multiplication, i.e. is it self*other or other*self, when each is viewed as a (passive) Lorentz transformation on the coordinates.